sphinx/source/docs/user_guide/tools.rst

```diff
@@ -561,6 +562,218 @@ shown below:
 .. bokeh-plot:: docs/user_guide/examples/tools_hover_custom_tooltip.py
     :source-position: above
 
+
+.. _userguide_tools_edit:
+
+Edit Tools
+----------
+
+The edit tools provide functionality for drawing and editing glyphs
+client-side by adding, modifying and deleting ``ColumnDataSource``
+data.
+
+All the edit tools share a small number of key bindings:
+
+* SHIFT: Allows adding to a selection or continue drawing
+* DELETE: Deletes the selected glyphs
+* ESC: Clear the selection
+
+.. note::
+
+   On some shortened keyboards (including MacBooks) the DELETE key is
+   missing and can be toggled by pressing FN + BACKSPACE
```

Comment:
I've never seen a macbook without a delete key (a key literally physically labeled "delete", am looking at one right now) , but I can't find a backspace on this MPB, so this note seems very confusing

Reply:
Right, so on my MPB I have to hit FN and the key literally labeled "delete" so I think this will be very confusing as is

Reply:
Worth clarifying this since it is confusing, the key labeled ``delete`` on a MacBook actually maps to the backspace key code, when you press Fn+ the key labeled "delete" it maps to the "delete" key code.

Reply:
Most users are not going to care or know about keycodes. Is there not a way re-map the key code to something uniform so that users can be spared from having to consider this at all?

Reply:
Because this will definitely generate repeated bug reports from mac users.

Reply:
I'd even suggest something else entirely, e.g. "x" to delete, if delete can't be "just made to work" on all platforms

Reply:
Any suggestion for a key? I originally simply had it mapped to backspace, but since key events are processed as long as the tool is active and backspace is a very common key to press it was too easy to accidentally delete your selection. How about shift-backspace?

Reply:
The same key does "just work" to delete backwards on all platforms, it's just that Mac decided to call it Delete, PCs decided to call it Backspace, and even worse, PCs use the label Delete for an entirely different key.  The key that deletes backwards is what deletes selected stuff in any drawing program or editor I've ever seen across platforms, so it's the key people will be trying regardless of any docs we write. Thus even though it's confusing to describe, I think it's much more discoverable than 'x' or any other key, so I think it should still work as it does now, just described as best we can.

Reply:
> The key that deletes backwards is what deletes selected stuff in any drawing program or editor I've ever seen across platforms, so it's the key people will be trying regardless of any docs we write. 

I agree! But that key on mac is just "delete", *not* "FN + delete". So I am suggesting to find a way to make "delete" work on mac, and everything the same on other platforms.

Reply:
>So I am suggesting to find a way to make just "delete" work on mac

I initially had it mapped to the backspace key code but didn't like it because that's such a common key to press and it's too easy to accidentally delete the current selection while editing a cell. Having it behave that way for OSX would still have that problem.

Reply:
Is that a big problem if undo works well?

Reply:
> I initially had it mapped to the backspace key code but didn't like it because that's such a common key to press and it's too easy to accidentally delete the current selection while editing a cell. Having it behave that way for OSX would still have that problem.

How is that not a problem on other platforms then? It's physically the same upper right key (called "delete" on mac and called "backspace" on others). If hitting bare "delete" on osx is to problematic, then it must also be the case that hitting bare "backspace" on others is also problematic for the exact same reason. If this is an issue because key bindings are global we can and probably should definitely check that the cursor is inside the plot before deleting anything.

Reply:
>If hitting bare "delete" on osx is to problematic, then it must also be the case that hitting bare "backspace" on others is also problematic for the exact same reason.

That's true, I thought you were suggesting using the delete key on OSX and the entirely different delete key on other machines.

>. If this is an issue because key bindings are global we can and probably should definitely check that the cursor is inside the plot before deleting anything.

If there's an easy way to do that I have no issue at all with using the backspace key code.

Reply:
> That's true, I thought you were suggesting using the delete key on OSX and the entirely different delete key on other machines.

Just to make sure I'm not being amibiguous, I am suggesting using the physically most upper, most right key, where "upper" is bounded by the row of number keys. 

> If there's an easy way to do that I have no issue at all with using the backspace key code.

I think we'd have to toggle a flag on canvas mouse enter/mouse leave events. I still don't understand how it can be too easy to hit the most upper right key on a mac (necessitating adding FN) but it's somehow not too easy to hit the same most upper right key on all other platforms? That logic of that does not make sense to me.

Reply:
>I still don't understand how it can be too easy to hit the most upper right key on a mac (necessitating adding FN) but it's somehow not too easy to hit the same most upper right key on all other platforms?

This is all very confusing. Fn+delete on OSX is equivalent to (i.e. emits the same key code as)  the actual dedicated delete key (located under the insert key) on full keyboards. That was a safer choice when it was capturing events globally because on OSX systems you'd have to press Fn+delete and on other systems the dedicated delete key (not the backspace). If we do detect whether you're hovering over the plot then backspace is fine.

Reply:
>I think we'd have to toggle a flag on canvas mouse enter/mouse leave events.

That worked, it now deletes if you press backspace (or the OSX key labelled "delete") while the mouse is within the frame.